src/Manifolds.jl

```diff
@@ -664,9 +668,9 @@ export get_basis,
 export AbstractDiffBackend,
     AbstractRiemannianDiffBackend,
     FiniteDifferencesBackend,
-    RiemannianONBDiffBackend,
-    RiemannianProjectionGradientBackend
-export diff_backend, diff_backend!, diff_backends
+    TangentDiffBackend,
+    RiemannianProjectionBackend
```

Comment:
The new names are nice but unfortunately this is breaking (so I'd recommend calling it 0.7.0)

Reply:
I think we even had one more thing we wanted to do for the next break? Maybe we should start having a dev branch of a branch for the next 0.x version to collect small breaking stuff...

But here, you are right. This is a breaking change (though it might not have been used before, still).

Reply:
One thing I want to do by the next breaking release was removing `ProductArray`, I'm not sure at the moment if there is anything else but we could track it somewhere.

Reply:
Ah, that's what I had in mind. maybe a dev or dev0.7 branch would be great for keeping track of those non.urgent but breaking changes (for a next breaking one that we want to actively do like this one)?

Reply:
I have never done a dev branch like that so I don't really know how much trouble would that be to maintain (compared to a github milestone for example).

Reply:
I have never used a milestone. My idea would have been that instead of just one PR for a new release like now, we collect a few PRs into a dev0.7 branch (all that are breaking for example) and wait for a few of them to be there – or one to be so interesting that we want to do a breaking PR now. with merging that the branch could also vanish, then it would just be a `dev0.7.0` branch.

Reply:
Well, there is no harm in trying that, at worst we would abandon the branch. Feel free to make that if you want :slightly_smiling_face: .

Reply:
Would you prefer milestones? My main point is, that I would like to avoid forgetting barking things that we have in mind to add for the next time we do a breaing release – like now with the removal of the type – I already had forgotten those details.

Reply:
I think either a milestone or an issue with a list would be just easier. You can take a look at, for example, `IncrementalInference.jl` to see how milestones work: https://github.com/JuliaRobotics/IncrementalInference.jl/milestones .

Reply:
I've made a milestone for us :slightly_smiling_face: .